rustls/src/client/hs.rs

```diff
@@ -232,8 +232,8 @@ fn emit_client_hello_for_retry(
         ClientExtension::CertificateStatusRequest(CertificateStatusRequest::build_ocsp()),
     ];
 
-    if let (ServerName::DnsName(sni_name), true) = (&input.server_name, config.enable_sni) {
-        exts.push(ClientExtension::make_sni(sni_name.borrow()));
+    if let (ServerName::DnsName(dns), true) = (&input.server_name, config.enable_sni) {
```

Comment:
If my feedback about adding a comment about why we do this `let` match on the `DnsName` variant made sense to you it could also be duplicated here.